### PR TITLE
numerous refactors for SluggabeBehavior class, fixes #10118

### LINKED ISSUE #10118: extending SluggableBehavior
I need a slug field for my model and it's going to be shown in url address. current `yii\helpers\Inflector::slug()` method does not produce what I need (it removes characters of my language) so I easily extended it and made needed changes.
The problem is there is no easy way to make SluggableBehavior use the new `slug()` method instead of default one! there is a 50-line-length method named [`getValue()`](https://github.com/yiisoft/yii2/blob/master/framework/behaviors/SluggableBehavior.php#L132) and among so many other things, it produces the real slug value to be saved. Am I supposed to extend the behavior and copy the whole method to edit just [1 line](https://github.com/yiisoft/yii2/blob/master/framework/behaviors/SluggableBehavior.php#L157) of it?
I think it should have a `protected generateSlug()` method which makes it easy to extend and alter slug creation logic. or is there another way I'm not aware of?

### PR DESCRIPTION
**Pull request by hesna**
it was made to fix #10118 but @cebe was way faster :+1: 
it still applies some refactors which I find useful.

### CODE REVIEW

**Comment by cebe** [comment]
Sorry for the late reply, could you explain why you think these refactorings are needed?
I currently do not really see the value of splitting out a lot of methods.

**Comment by hesna** [comment]
@cebe no problem. actually I wouldn't touch it if it wasn't for #10118. while fixing that issue, I find it better to refactor the whole class. 
It's just about readability and thus easier maintenance. I know code quality is a relative issue and one may not agree with me, but I find current implementation better than the old one. just having a 50-line method bothers me actually :) (btw it seems [Scrutinizer](https://scrutinizer-ci.com/g/yiisoft/yii2/inspections/fa83399a-d252-461e-b7c6-26d7d6bc721b) likes the new version better)

File: framework/behaviors/SluggableBehavior.php [hunk 1]
```diff
@@ -131,49 +131,49 @@ public function init()
...
-        return $slug;
+
+        return false;
     }
 
     /**
+     * checks if owner model already has a slug value or not.
+     * @return boolean
+     */
+    private function ownersSlugValueIsEmpty()
```
Comment (SilverFire) [inline framework/behaviors/SluggableBehavior.php]: I think this method is redundant.

**Comment by SilverFire** [comment]
I go for this approach because is is more OOP-style, the responsibilities are separated to dedicated methods.

We need to improve PHPDocs to these methods and fix code style, then it's ready to merge.
@cebe may I reassign this PR to myself and finish it?

**Comment by hesna** [comment]
@SilverFire Hi, I tried to fix some coding style problems, please let me know if there are more. sorry about PHPDocs, I'm not good at that.

**Comment by SilverFire** [comment]
Merged with minor adjustments, thank you!
https://github.com/yiisoft/yii2/commit/6936c41cf1a047a2e35b50175d9bf83619f3ab2a

**Pull request closed without merging**